### PR TITLE
Added parallel implementation of LogisticRegression::compute_gradient.

### PR DESCRIPTION
**Pull request by BloodAxe**
### What does this PR change?
This PR adds multi-threaded computation of gradients in LogisticRegression::compute_gradient method.
Nothing else changed.

### CODE REVIEW

**Comment by BloodAxe** [comment]
I'm not quite sure how to react on this failing build, since this PR does not affect OpenCL module at all.

**Comment by alalek** [comment]
Looks good to me! :+1:

**Comment by BloodAxe** [comment]
Thanks @alalek ! For many approvals required for PR to merge?

File: modules/ml/src/lr.cpp [hunk 1]
```diff
@@ -362,6 +362,42 @@ double LogisticRegressionImpl::compute_cost(const Mat& _data, const Mat& _labels
...
 
+struct LogisticRegressionImpl_ComputeDradient_Impl : ParallelLoopBody
+{
+    const Mat* data;
+    const Mat* theta;
+    const Mat* pcal_a;
+    Mat* gradient;
+    double lambda;
+
+    LogisticRegressionImpl_ComputeDradient_Impl(const Mat& _data, const Mat &_theta, const Mat& _pcal_a, const double _lambda, Mat & _gradient)
```
Comment (sturkmen72) [inline modules/ml/src/lr.cpp]: LogisticRegressionImpl_ComputeDradient_Impl is it a typo ?

**Pull request merged**